Fill rois

Rdornier opened the pull request:
Hello, 

This PR should fiy the issue #485 about ROI filling.
If you load ROIs from OMERO, the fill-color is read correctly but not the transparency. I wasn't able to get it from the imported json object. So, the current transparency is set to the loaded ROI.

Rémy.

will-moore commented:
NB: I'm away for holidays after tomorrow so it may be a while before I can look at this again...

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#118](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/118/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/118/consoleText) for more details.
Possible conflicts:
  - PR #549 will-moore 'Inset feature'
    - src/js/shape_editor/rect.js

--conflicts

will-moore commented:
I've removed the `include` label to avoid the "Conflicting PR" messages every day. #549 isn't quite ready to merge just yet so I'm afraid we'll have to wait a bit.

Rdornier commented:
Hi @will-moore 

Thanks for the feedback and the suggestion

> You shouldn't commit any files under omero_figure/static/figure/... - They should be ignored by .gitignore.

It has been included in the gitignore

I added the rounding with 2 decimales

> NB: I'm away for holidays after tomorrow so it may be a while before I can look at this again...

No worries, enjoy your holidays !

> I've removed the include label to avoid the "Conflicting PR" messages every day.

OK

Rémy

will-moore commented:
@Rdornier This is looking pretty good now. I haven't tested the PDF/Tiff export yet. I seem to remember there were some issues there, which is why I removed the fillOpacity code from the ShapeEditor code when I ported it over. I'll give that a go anyway...
I had a go at fixing the toolbar wrapping above. See last commit at https://github.com/will-moore/figure/commits/fill-rois/ which gives the toolbar the whole top row of the dialog.

Rdornier commented:
Hi @will-moore 

>  I haven't tested the PDF/Tiff export yet. I seem to remember there were some issues there

True, I have to check as well
I tried to build the app, as described in the readme but it gives me this output

```
$ npm run build

 >omero-figure@0.0.0 build
 >vite build --emptyOutDir && ./deploy_build.sh

vite v3.2.10 building for production...
transforming...
✓ 139 modules transformed.
rendering chunks...
../omero_figure/static/omero_figure/assets/luts_10.863786e7.png   18.12 KiB
../omero_figure/static/omero_figure/index.html                    63.62 KiB
../omero_figure/static/omero_figure/assets/index.3fe885b4.css     251.65 KiB / gzip: 42.17 KiB
../omero_figure/static/omero_figure/assets/index.05660f14.js      659.28 KiB / gzip: 206.62 KiB
../omero_figure/static/omero_figure/assets/index.05660f14.js.map  2277.14 KiB
'.' is not recognized as an internal or external command,
operable program or batch file.
```

I don't know if this is expected, probably not. But then, how can I use this output to test the save/export ?

> I had a go at fixing the toolbar wrapping above. See last commit at https://github.com/will-moore/figure/commits/fill-rois/ which gives the toolbar the whole top row of the dialog.

Ok

Rdornier commented:
I've updated the Figure_to_pdf script acording to what I understood from the code and from what was missing to fit with the new feature.
I didn't manage yet to make and use the figure build, so, I didn't test if the changes worked or not.

will-moore commented:
This is what I see:

```
$ npm run build

> omero-figure@0.0.0 build
> vite build --emptyOutDir && ./deploy_build.sh

vite v3.2.10 building for production...
✓ 139 modules transformed.
../omero_figure/static/omero_figure/assets/luts_10.863786e7.png   18.12 KiB
../omero_figure/static/omero_figure/index.html                    61.77 KiB
../omero_figure/static/omero_figure/assets/index.676c7de2.css     251.38 KiB / gzip: 42.09 KiB
../omero_figure/static/omero_figure/assets/index.c141c1a9.js      656.93 KiB / gzip: 206.49 KiB
../omero_figure/static/omero_figure/assets/index.c141c1a9.js.map  2418.04 KiB

(!) Some chunks are larger than 500 KiB after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/guide/en/#outputmanualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
Deploying built resources to plugin directory...
```

It looks like the JS build worked OK. I guess you see the various assets under `../omero_figure/static/omero_figure/assets/` ?
Can you just try running the `./deploy_build.sh` script manually to copy those assets into the correct place?

```
$ ./deploy_build.sh
Deploying built resources to plugin directory...
```

Rdornier commented:
>  I guess you see the various assets under ../omero_figure/static/omero_figure/assets/ ?

Yes, I do. 

> Can you just try running the ./deploy_build.sh script manually to copy those assets into the correct place?

It works, thanks. Now, I'm able to fully build and test

will-moore commented on omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py:
```diff
@@ -771,7 +777,9 @@ def draw_ellipse(self, shape):
         rgba = ShapeToPdfExport.get_rgba_int(shape['strokeColor'])
         ellipse_draw.ellipse((0, 0, width, height), fill=rgba)
-        rgba = self.get_rgba_int(shape.get('fillColor', '#00000000'))
+        r,g,b,a = self.get_rgba_int(shape.get('fillColor', '#00000000'))
```
This line (and others above) are causing flake8 build failures:
```
flake8.main.application   MainProcess    652 INFO     Found a total of 99 violations and reported 9
./omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py:651:10: E231 missing whitespace after ','
./omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py:651:12: E231 missing whitespace after ','
./omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py:651:14: E231 missing whitespace after ','
./omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py:709:10: E231 missing whitespace after ','
./omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py:709:12: E231 missing whitespace after ','
./omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py:709:14: E231 missing whitespace after ','
./omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py:780:10: E231 missing whitespace after ','
./omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py:780:12: E231 missing whitespace after ','
./omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py:780:14: E231 missing whitespace after ','
```

will-moore commented:
The integration tests are failing to export figure with error:

```
alpha = int(float(shape[\'fillOpacity\'])*255)\nKeyError: \'fillOpacity\'\n'
```

Looking at the code, I realise that the export script supports fillColor already (and possibly opacity if the colour is rgba). This is additional functionality added by Glencoe software as they use the export script independently from the OMERO.figure app. See https://github.com/ome/omero-figure/pull/306 

In order to fix the error above, and to not break the existing `rgba` logic if `fillOpacity` isn't given, you can do this everywhere:

```
        r, g, b, a = self.get_rgba_int(shape.get('fillColor', '#00000000'))
        if 'fillOpacity' in shape:
            a = int(float(shape['fillOpacity'])*255)
        rgba = (r, g, b, a)
```

Rdornier commented:
Ok, I fixed it so that the rgba logic is not broken

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#170](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/170/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/170/consoleText) for more details.
Possible conflicts:
  - PR #549 will-moore 'Inset feature'
    - src/js/shape_editor/rect.js
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py
  - PR #560 will-moore 'Max projection bytes'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#171](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/171/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/171/consoleText) for more details.
Possible conflicts:
  - PR #549 will-moore 'Inset feature'
    - src/js/shape_editor/rect.js
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py
  - PR #560 will-moore 'Max projection bytes'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#172](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/172/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/172/consoleText) for more details.
Possible conflicts:
  - PR #549 will-moore 'Inset feature'
    - src/js/shape_editor/rect.js
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py
  - PR #560 will-moore 'Max projection bytes'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#173](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/173/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/173/consoleText) for more details.
Possible conflicts:
  - PR #549 will-moore 'Inset feature'
    - src/js/shape_editor/rect.js
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py
  - Upstream changes
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#174](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/174/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/174/consoleText) for more details.
Possible conflicts:
  - PR #549 will-moore 'Inset feature'
    - src/js/shape_editor/rect.js
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#175](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/175/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/175/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#176](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/176/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/176/consoleText) for more details.
Possible conflicts:
  - PR #549 will-moore 'Inset feature'
    - src/js/shape_editor/rect.js
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#177](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/177/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/177/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py
  - Upstream changes
    - src/js/shape_editor/rect.js

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#178](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/178/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/178/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py
  - Upstream changes
    - src/js/shape_editor/rect.js

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#179](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/179/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/179/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py
  - Upstream changes
    - src/js/shape_editor/rect.js

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#180](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/180/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/180/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#181](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/181/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/181/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#182](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/182/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/182/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#183](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/183/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/183/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#184](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/184/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/184/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#185](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/185/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/185/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#186](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/186/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/186/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#187](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/187/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/187/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#188](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/188/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/188/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#189](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/189/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/189/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#190](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/190/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/190/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#191](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/191/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/191/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#192](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/192/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/192/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#193](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/193/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/193/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#194](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/194/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/194/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#195](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/195/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/195/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#196](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/196/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/196/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#197](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/197/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/197/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#198](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/198/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/198/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#199](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/199/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/199/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#200](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/200/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/200/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#201](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/201/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/201/consoleText) for more details.
Possible conflicts:
  - PR #555 will-moore 'Points in shape editor'
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#202](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/202/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/202/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-plugins-push#203](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/203/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-plugins-push/203/consoleText) for more details.
Possible conflicts:
  - Upstream changes
    - omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

--conflicts

will-moore commented:
Thanks for the updates on this and other PRs. Just a heads up that I'm pretty tied up over the next week or so - Afraid I might not get time to review etc for a bit...

will-moore commented:
![Screenshot 2024-10-22 at 16 42 10](https://github.com/user-attachments/assets/05bb30ab-634b-46c7-b21e-d3845700c53d)

A couple of issues:
 - With Points PR now merged, when I try to select points on the Image in the ROI dialog, I get a console error:
 ```
shape_manager.js:679 Uncaught TypeError: l.getFillColor is not a function
    at shape_manager.js:679:27
```
 - It seems that when I pick a Shape loaded from OMERO (with fill Color rgba `255, 255, 255, 0`) and "Add" it to the Figure panel, it seems that the Opacity that is applied to it is the current value of the Fill-Opacity select option. E.g. here the current value is `0.20` and when I add the Ellipse, that's the Opacity it gets (not the 0 opacity loaded from OMERO)

<img width="930" alt="Screenshot 2024-10-22 at 22 36 51" src="https://github.com/user-attachments/assets/ee9b0572-1d01-4355-8eff-4cbbb1a0d890">

 - It would be nice to fix the toolbar wrapping. Here's a commit where I tried this above - hopefully this will still work: https://github.com/will-moore/figure/commit/7da63a6883bb074cbad5db1fa00c445ac53a5860

Rdornier commented:
Hi @will-moore 

Thanks for your feedback.

> The drop-down for choosing Opacity above looks strange with the 2 decimal places. E.g. 0, 0.1, 0.2, 0.3.. would look better.

Ok, I'll correct it

> With Points PR now merged, when I try to select points on the Image in the ROI dialog, I get a console error:

I think this is normal, as Point ROI doesn't have yet the methods to get fill etc. I'll add them.

> It seems that when I pick a Shape loaded from OMERO (with fill Color rgba 255, 255, 255, 0) and "Add" it to the Figure panel, it seems that the Opacity that is applied to it is the current value of the Fill-Opacity select option. E.g. here the current value is 0.20 and when I add the Ellipse, that's the Opacity it gets (not the 0 opacity loaded from OMERO)

Yes, it is the current behavior. I don't remember why I didn't read the opacity from OMERO. I'll check it.

I'm a bit busy this and next week. So, I'll most probably do the corrections at the beginning of November.

Rémy.

will-moore reviewed (approved):
Functionality looking good, thanks!

![Screenshot 2024-11-11 at 16 25 26](https://github.com/user-attachments/assets/2ad32bb3-d8b2-4f24-81c9-7b9e3d3d2145)
